src/api/DesktopAgent.ts

```diff
@@ -547,8 +548,27 @@ export interface DesktopAgent {
    */
   getAppMetadata(app: AppIdentifier): Promise<AppMetadata>;
 
+  /**
+   * Use by getAgent() to validate app identity. Apps should not call this function directly.
+   *
+   * See getAgent.md for instructions.
+   *
+   * Either appId, appDUrl, or both may be provided. This logic is covered int supported_platforms.md.
+   *
+   * This function is optional but recommended.
+   */
+  validateAppIdentity?({
```

Comment:
I don't understand why this function would be here.  This is not something an app would call.  Is this a mistake?

Reply:
Do you remember the conversations about there being a baked-in assumption that apps don't navigate? Thats always been incorrect, but most DA implementations don't do much about it. FDC3 for the Web is/has to be tighter on that topic - but there isn't an equivalent solution in containers. When we were using appD records, we needed a function call for equivalence so that the app could indicate its identity - later could be extended to use async encryption to confirm it.

Needs updating to the URL based approach. Still needs an API call however, due to the identity URL.

Reply:
I don't get it.  Who call this, when and why?  If it's part of the handshake process there shouldn't be a method for it - we have getAgent to handle this

Reply:
`getAgent()` could call it if it exists as an API call for convenience. It's not had a huge amount of discussion lately compared to the rest of the proposal (as its parity for the containers to keep up with the FDC3 for the web)

Reply:
P.S. it'd have to be exposed on the API for `getAgent()` to call it, as thats all you've got access to for container-based Desktop Agents

Reply:
>P.S. it'd have to be exposed on the API for getAgent() to call it, as thats all you've got access to for container-based Desktop Agents

We don't need to validate app identity for electron based apps... otherwise this method would already exist!

> getAgent() could call it if it exists as an API call for convenience. It's not had a huge amount of discussion lately compared to the rest of the proposal (as its parity for the containers to keep up with the FDC3 for the web)

The `DesktopAgent` interface is client-facing.  (i.e. for apps to use), so it's wrong to have it here for this reason.  So that's just adding noise to an already over-complicated API which people complain to me regularly about.

Second, having it here is exposing implementation decisions of FDC3 on the web, which means that electron / .net / Java etc. will all have to implement a completely pointless method call.

Reply:
@robmoffat that function is purely for container-based DesktopAgents. Theres no DA-proxy code involved in it - the same thing is handled in the WCP methods there. 

With containers, `getAgent()` returns the window.fdc3 element the container injects, then the app or `getAgent` should call `validateIdentity` to do the same identity validation that the DA proxy does with WCP messages.

In the meetings on the subject, we discussed the assumption that most FDC3 implementations use that apps don't navigate. It's not correct. Containers can track the URLs, but they never havd any formal way to relate that back to an appD record and updated identity. We've had to work that out for fdc3 for the web (initially proposed to nominate an appD record - you recently had us switch to using a URL, and we had to allow for an app nominated 'identityUrl' as apps may put state in or otherwise use their URLs in various way). There ought to be some way for container-based DAs to do the same thing. However, the only interface we have to those is the public API.

The meetings agreed to introduce this as optional / @experimental initially to allow containers to explore implementing it and having parity with fd3 for the web based DAs on identity.

Reply:
Updated and simplified the function definition and marked it @experimental. However, I'm realizing that its not going to be a huge amount of use without it being required, which will need more discussion with the SWG. It may need separating out into 2.3.

I don't currently see another way we can do this containers - it has to be an API function as thats the only interface. We need someway for the app to express its identity - and we've moved away from appD records, and hence need to use the identityUrl approach (which isn't 100% ideal due to the need to match to URLs in the appD records)...

Reply:
We should talk about this - I don't really understand what you're trying to achieve here and I don't think GitHub comments is helping much!   Are you coming to the implementor's meeting this afternoon?  We can talk about it then